### ISSUE TITLE
Argument 1 passed to SebastianBergmann\CodeCoverage\ProcessedCodeCoverageData::merge() must be an instance of SebastianBergmann\CodeCoverage\ProcessedCodeCoverageData

### ISSUE THREAD

**Issue by NeilJ247** [Codeception/Codeception#6042]
#### What are you trying to achieve?

Run code coverage with latest versions of codeception, phpunit and xdebug v3

#### What do you get instead?

An error when merging coverage data:

```
Fatal error: Uncaught TypeError: Argument 1 passed to SebastianBergmann\CodeCoverage\ProcessedCodeCoverageData::merge() must be an instance of SebastianBergmann\CodeCoverage\ProcessedCodeCoverageData, array given, called in /var/www/html/vendor/phpunit/php-code-coverage/src/CodeCoverage.php on line 319 and defined in /var/www/html/vendor/phpunit/php-code-coverage/src/ProcessedCodeCoverageData.php on line 135




FATAL ERROR. TESTS NOT FINISHED.
Uncaught TypeError: Argument 1 passed to SebastianBergmann\CodeCoverage\ProcessedCodeCoverageData::merge() must be an instance of SebastianBergmann\CodeCoverage\ProcessedCodeCoverageData, array given, called in /var/www/html/vendor/phpunit/php-code-coverage/src/CodeCoverage.php on line 319 and defined in /var/www/html/vendor/phpunit/php-code-coverage/src/ProcessedCodeCoverageData.php:135
Stack trace:
#0 /var/www/html/vendor/phpunit/php-code-coverage/src/CodeCoverage.php(319): SebastianBergmann\CodeCoverage\ProcessedCodeCoverageData->merge(Array)
#1 /var/www/html/vendor/codeception/codeception/src/Codeception/Coverage/SuiteSubscriber.php(106): SebastianBergmann\CodeCoverage\CodeCoverage->merge(Object(SebastianBergmann\CodeCoverage\CodeCoverage))
#2 /var/www/html/vendor/codeception/codeception/src/Codeception/Coverage/Subscriber/LocalServer.php(126): Codeception\Coverage\SuiteSubscriber->mergeToPrint(Object(SebastianBergmann\CodeCoverage\CodeCoverage))
#3 /var/www/html/vendor/symfony/event-dispatcher/EventDispa 
in /var/www/html/vendor/phpunit/php-code-coverage/src/ProcessedCodeCoverageData.php:135

```
> Provide test source code if related

```php
<?php

//phpcs:ignore PSR1.Classes.ClassDeclaration.MissingNamespace
class ApiGetRequestToAllPhoneCodesCest
{
    public function tryApi(ApiTester $I)
    {
        $I->sendGET('/country-codes/phone');
        $I->seeResponseCodeIs(200);
        $I->canSeeResponseIsJson();
        $I->seeResponseContainsJson([
            'BD' => '880',
            'LA' => '856',
        ]);
    }
}
```

### Details

* Codeception version: 4.1.12
* PHP Version: 7.4
* Operating System: Linux Alpine
* Installation type: Composer
* List of installed packages: 
        "codeception/c3": "^2.6",
        "codeception/codeception": "^4.1",
        "codeception/module-asserts": "^1.3",
        "codeception/module-phpbrowser": "^1.0",
        "codeception/module-rest": "^1.2",
        "php-coveralls/php-coveralls": "^2.0",
        "phpunit/phpunit": "^9.4"
* Suite configuration:

```yml
paths:
    tests: tests
    output: var/_codeception/_output
    data: var/_codeception/_data
    support: var/_codeception/_support
actor_suffix: Tester
params:
    - .env
extensions:
    enabled:
        - Codeception\Extension\RunFailed
coverage:
    enabled: true
    include:
        - src/Application/*
        - src/Domain/*
settings:
    shuffle: false
    lint: true
```

**Comment by aymanalzarrad** [comment]
This is caused by an update to `phpunit/php-code-coverage` and it's partially discussed here: #5949 

The only solution that is working for me currently is to downgrade to `phpunit/php-code-coverage:^8.0`. 

Note: the lock will prevent the downgrade so you will have to remove the other packages of `codeception/*` and `phpunit/*` before downgrading.
```
// Remove codeception/* and phpunit/*
$ composer remove "codeception/*" "phpunit/*" --dev

// Downgrade phpunit/php-code-coverage
$ composer require "phpunit/php-code-coverage:^8.0" --dev

// Re-require the other packages
$ composer require codeception/c3 codeception/codeception codeception/module-asserts codeception/module-phpbrowser codeception/module-rest php-coveralls/php-coveralls phpunit/phpunit
```

**Comment by Naktibalda** [comment]
@aymanalzarrad @NeilJ247 I can't reproduce this issue - everything just works for me.
My guess would be that either you got an old version of c3.php or your c3.php includes an old version of php-code-coverage library.

Would you be able to provide contents of `tests/_output/c3tmp/codecoverage.serialized` file?

**Comment by aymanalzarrad** [comment]
@Naktibalda The issue seems to be occurring only when updating a pre-existing setup and not on a freshly bootstrapped one.

I will try to get back to you with the serialized file after the holidays. In the meanwhile, I wish you a happy one :)

**Comment by aymanalzarrad** [comment]
I have looked a bit more into this and while doing so I noticed that the files in the `_output` directory weren't being cleared correctly.

My guess is that something (Not sure what) is preventing a proper clearing of pre-existing serialised files from previous tests runs after an update.

However, cleaning the directory manually and re-running tests everything runs and works as it should, so technically I can't reproduce the error anymore.
